pkgs/development/python-modules/bdsim/default.nix

```
@@ -0,0 +1,84 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+  oldest-supported-numpy,
+  setuptools,
+  ansitable,
+  matplotlib,
+  numpy,
+  progress,
+  scipy,
+  spatialmath-python,
+  coverage,
+  flake8,
+  pytest,
+  pytest-timeout,
+  pytest-xvfb,
+  sympy,
+  pillow,
+  pyqt5,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "bdsim";
+  version = "1.1.1";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-BK1lzCwSu9NZZmTaSNyDWPm47bepaQmD0MrbaajDw1g=";
+  };
+
+  nativeBuildInputs = [
```

```suggestion
  build-system = [
```
Could you use `build-system` instead of `nativeBuildInputs`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters